src/utilities/strategy-helpers/deployment-helper.ts

```diff
@@ -112,19 +112,36 @@ async function checkManifestStability(kubectl: Kubectl, resources: Resource[]):
     await KubernetesManifestUtility.checkManifestStability(kubectl, resources);
 }
 
-function annotateResources(files: string[], kubectl: Kubectl, resourceTypes: Resource[], allPods: any) {
+async function annotateAndLabelResources(files: string[], kubectl: Kubectl, resourceTypes: Resource[], allPods: any) {
+    const workflowFilePath = await getWorkflowFilePath(TaskInputParameters.githubToken);
+    const annotationKeyLabel = models.getWorkflowAnnotationKeyLabel(workflowFilePath);
+    annotateResources(files, kubectl, resourceTypes, allPods, annotationKeyLabel);
+    labelResources(files, kubectl, annotationKeyLabel);
+}
+
+function annotateResources(files: string[], kubectl: Kubectl, resourceTypes: Resource[], allPods: any, annotationKey: string) {
     const annotateResults: IExecSyncResult[] = [];
-    annotateResults.push(annotateNamespace(kubectl, TaskInputParameters.namespace));
-    annotateResults.push(kubectl.annotateFiles(files, models.workflowAnnotations, true));
+    const lastSuccessSha = getLastSuccessfulRunSha(kubectl, TaskInputParameters.namespace, annotationKey);
+    let annotationKeyValStr = annotationKey + '=' + models.getWorkflowAnnotationsJson(lastSuccessSha);
+    annotateResults.push(kubectl.annotate('namespace', TaskInputParameters.namespace, [annotationKeyValStr], true));
+    annotateResults.push(kubectl.annotateFiles(files, [annotationKeyValStr], true));
     resourceTypes.forEach(resource => {
         if (resource.type.toUpperCase() !== models.KubernetesWorkload.pod.toUpperCase()) {
-            annotateChildPods(kubectl, resource.type, resource.name, allPods)
+            annotateChildPods(kubectl, resource.type, resource.name, annotationKeyValStr, allPods)
                 .forEach(execResult => annotateResults.push(execResult));
         }
     });
     checkForErrors(annotateResults, true);
 }
 
+function labelResources(files: string[], kubectl: Kubectl, label: string) {
+    let workflowName = process.env.GITHUB_WORKFLOW;
+    workflowName = workflowName.startsWith('.github/workflows/') ?
+        workflowName.replace(".github/workflows/", "") : workflowName;
+    const labels = [`workflowFriendlyName=${workflowName}`, `workflow=${label}`];

```

Comment:
Why the camel casing?

Reply:
That has been suggested by Rachit to be part of the label selector. Any suggestions?

Reply:
`workflow-friendly-name` abiding to k8s casing